src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeLikelihoodCalculatorDRAGEN.java

```diff
@@ -0,0 +1,487 @@
+                            * ((Math.min(depthForGenotyping, maxEffectiveDepthForHetAdjustment) * 1.0) / depthForGenotyping));
+                    outputArray[indexForGT] = Math.max(outputArray[indexForGT], adjustedBestModel + localBestModel[1]);
+
+                    if (HaplotypeCallerGenotypingDebugger.exists()) {
+                        HaplotypeCallerGenotypingDebugger.println("best FRD likelihoods: "+localBestModelScore+" P(F) score used: "+localBestModel[1]+"  use MaxEffectiveDepth: "+maxEffectiveDepthForHetAdjustment);
+                        HaplotypeCallerGenotypingDebugger.println("Using array index "+closestGTAlleleIndex+" for mixture gt with likelihood of "+log10LikelihoodsForPloyidyModel+" adjusted based on depth: "+depthForGenotyping);
+                        HaplotypeCallerGenotypingDebugger.println("p_rG_adj : "+adjustedBestModel);
+                    }
+                } else {
+                    outputArray[indexForGT] = Math.max(outputArray[indexForGT], localBestModel[0]);
+                }
+
+            }
+
+        }
+
+
+        return outputArray;
+    }
+
+
+    /**
+     * @param positionSortedReads read containers to use for genotyping
+     * @param predicate predicate used to select the correct orientation combination for reads when genotyping
+     * @param offsetForReadLikelihoodGivenAlleleIndex offset corresponding to the Error Allele in the reads likelihoods object array
+     * @param readLikelihoodsForGT reads likelihoods for Genotype array table
+     * @param criticalThresholdsSorted critical thresholds to use for this orientation combination
+     * @return two doubles, index 0 is the frd score and the second is log p(F()) score used to adjust the score
+     */
+    private double[] computeFRDModelForStrandData(final List<DRAGENGenotypesModel.DragenReadContainer> positionSortedReads, final Predicate<DRAGENGenotypesModel.DragenReadContainer> predicate,
+                                                  final int offsetForReadLikelihoodGivenAlleleIndex, final double[] readLikelihoodsForGT, final List<Double> criticalThresholdsSorted) {
+        if (positionSortedReads.isEmpty()) {
+            return new double[]{Double.NEGATIVE_INFINITY, 0};
+        }
+
+        int counter = 0;
+        double maxLpspi = Double.NEGATIVE_INFINITY;
+        double lpfApplied = 0;
+
+        for (final Double logProbFAllele : criticalThresholdsSorted) {
+            double fAlleleProbRatio = 0.0;
+            double fAlleleProbDenom = 0.0;
+            double localMaxLpspi = Double.NEGATIVE_INFINITY;
+
+            // iterate over the reads to compute the foreign allele alpha to use for genotyping with FRD
+            for (final DRAGENGenotypesModel.DragenReadContainer container : positionSortedReads) {
+                // Ignore reads that were disqualified by the HMM
+                if (container.wasFilteredByHMM()) {
```

Comment:
The filtering was added at this layer for a reason, namely DRAGEN includes filtered reads from the HMM for selecting threshold for FRD priors to use for seperating reads into "error" and "not error" partitions but since those reads have no HMM scores they don't actually contribute anything to the genotype scores here. This means that at the point where I call computeCriticalValues() I needed those hmm failing reads now I want to ignore them (because they have no more likelihoods).  I could have filtered these reads but it would involve another layer of construction that I didn't want to pay the overhead for so I make the check here. Would you recommend changing that?